src/main/resources/com/microsoft/jenkins/azuread/AzureSecurityRealm/config.jelly

```diff
@@ -6,34 +6,47 @@
 
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
     <f:block>
-        <f:entry title="Client ID"  field="clientId" help="/plugin/azure-ad/help/help-client-id.html" >
+        <f:entry title="Client ID"  field="clientId" >
             <f:textbox />
         </f:entry>
 
-        <f:entry title="Client Secret" field="clientSecret" help="/plugin/azure-ad/help/help-client-secret.html">
-            <f:password />
+        <f:entry title="Authentication Type">
+            <f:radioBlock name="credentialType" value="Secret" title="Client Secret" help="${null}"
```

Comment:
radioBlock appears to be expecting a descriptor and as one isn't passed it was inheriting the one for this page giving a confusing help message, hacked it by passing `${null}`

Alternatively could add descriptors for auth type,
OR migrate to credentials API that accepts either a String or certificate credential which would probably be cleaner, and then checks would be be done based on instanceof for the resulting credential

(not a blocker can leave)

Reply:
Great, tested as well it looks good.